File: mbflow-ui/src/types/trigger.ts
export type TriggerType = "manual" | "schedule" | "webhook" | "event";
export type TriggerStatus = "enabled" | "disabled";

export interface TriggerConfig {
  // Schedule config
  cron?: string;
  timezone?: string;

  // Webhook config
  webhook_path?: string;
  http_method?: string;
  auth_type?: string;
  auth_config?: Record<string, any>;

  // Event config
  event_type?: string;
  event_filter?: Record<string, any>;

  [key: string]: any;
}

export interface Trigger {
  id: string;
  workflow_id: string;
  workflow_name?: string;
  name: string;
  description?: string;
  type: TriggerType;
  status: TriggerStatus;
  config: TriggerConfig;
  metadata?: Record<string, any>;
  created_at: string;
  updated_at: string;
  last_triggered_at?: string;
  next_trigger_at?: string;
}

export interface TriggerCreateRequest {
  workflow_id: string;
  name: string;
  description?: string;
  type: TriggerType;
  status?: TriggerStatus;
  config: TriggerConfig;
  metadata?: Record<string, any>;
}

export interface TriggerUpdateRequest {
  name?: string;
  description?: string;
  status?: TriggerStatus;
  config?: TriggerConfig;
  metadata?: Record<string, any>;
}

export interface TriggerListParams {
  workflow_id?: string;
  type?: TriggerType;
  status?: TriggerStatus;
  page?: number;
  limit?: number;
}

File: backend/sdk/go/models/trigger.go
package models

import "time"

// TriggerType represents the mechanism that initiates a workflow execution.
type TriggerType string

const (
	TriggerTypeManual   TriggerType = "manual"
	TriggerTypeCron     TriggerType = "cron"
	TriggerTypeWebhook  TriggerType = "webhook"
	TriggerTypeEvent    TriggerType = "event"
	TriggerTypeInterval TriggerType = "interval"
)

// Trigger represents a workflow trigger configuration.
type Trigger struct {
	ID          string         `json:"id"`
	WorkflowID  string         `json:"workflow_id"`
	Name        string         `json:"name"`
	Description string         `json:"description,omitempty"`
	Type        TriggerType    `json:"type"`
	Config      map[string]any `json:"config"`
	Enabled     bool           `json:"enabled"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	LastRun     *time.Time     `json:"last_run,omitempty"`
	NextRun     *time.Time     `json:"next_run,omitempty"`
	Metadata    map[string]any `json:"metadata,omitempty"`
}
